guides/Circle CI 2.md

```diff
@@ -0,0 +1,195 @@
+# Circle CI 2.0 configurations for Rails project
+
+## Basic Concepts
+
+CircleCI 2.0 operates on a differently structurized YAML file located under `.circleci/config.yml`.
+
+It's safe to say that every config should contain at least three keywords: `version, jobs`.
+Initially, it should start looking something like this:
+```
+---
+version: 2
+jobs:
+    build:
+        working_directory: ~/your-app-name
+    steps:
+        - checkout
+```
+This intital setup is not going to work. But it’s the basic shell of the config file we’ll need.
+
+## Basic Configurations.
+
+### Select image for Docker executor
+Which will compose the environment from a set of Docker images [CircleCI provide set of pre-built images that we’ll use. ](https://hub.docker.com/r/circleci/ruby/tags/)
```

Comment:
The docker image will help to compose a consistent environment for each build. There are a set of [prebuilt images](https://hub.docker.com/r/circleci/ruby/tags/) to choose from, which cover different ruby versions and other common requirements like browser and/or JavaScript.